Makefile.am

```diff
@@ -1,278 +1,135 @@
-test_unit_TPMU_marshal_SOURCES = test/unit/TPMU-marshal.c
-endif # UNIT
+    RELEASE.md
+
+# TODO: To be removed in the next commit
+EXTRA_DIST += \
+    lib/debug_config.site
+
+# Generate the AUTHORS file from git log
+AUTHORS :
+	$(AM_V_GEN)git log --format='%aN <%aE>' | grep -v 'users.noreply.github.com' | sort | \
+	    uniq -c | sort -nr | sed 's/^\s*//' | cut -d" " -f2- > $@
+EXTRA_DIST += AUTHORS
+CLEANFILES += AUTHORS
+
+# pkg-config setup. pc-file declarations happen in the corresponding modules
+pkgconfigdir          = $(libdir)/pkgconfig
+nodist_pkgconfig_DATA =
+CLEANFILES += $(nodist_pkgconfig_DATA)
+
+%.pc : %.pc.in
+	$(AM_V_GEN)$(call make_parent_dir,$@) && \
+	sed -e "s,[@]VERSION[@],$(PACKAGE_VERSION),g; \
+	        s,[@]libdir[@],$(libdir),g; \
+	        s,[@]includedir[@],$(includedir),g;" $^ > $@
 
+### Internal utility library
+libutil = libutil.la
+noinst_LTLIBRARIES += $(libutil)
 libutil_la_CFLAGS = $(AM_CFLAGS)
 libutil_la_SOURCES = log/log.c log/log.h tcti/tcti.c tcti/tcti.h
 
+### TCG TSS Marshaling/Unmarshalling spec library ###
+libmarshal = marshal/libmarshal.la
+lib_LTLIBRARIES += $(libmarshal)
+
 if HAVE_LD_VERSION_SCRIPT
 marshal_libmarshal_la_LDFLAGS = -Wl,--version-script=$(srcdir)/lib/libmarshal.map
 endif # HAVE_LD_VERSION_SCRIPT
 marshal_libmarshal_la_LIBADD = $(libutil)
 marshal_libmarshal_la_SOURCES = $(MARSHAL_SRC)
 
-sysapi_libsapi_la_LIBADD  = $(libmarshal)
-sysapi_libsapi_la_SOURCES = $(SYSAPI_C) $(SYSAPI_H) $(SYSAPIUTIL_C) \
-    $(SYSAPIUTIL_H)
+nodist_pkgconfig_DATA += lib/marshal.pc
+EXTRA_DIST += \
+    lib/libmarshal.map \
+    lib/marshal.pc.in
 
-if ESAPI
-esapi_libesapi_la_CFLAGS  = $(AM_CFLAGS) -Wno-unused-variable -Wno-unused-label \
-                            -I. -I$(srcdir)/esapi/esapi -I$(srcdir)/include/esapi \
-                            -I$(srcdir)/esapi/esapi_util \
-                            -DESYS_TCTI_DEFAULT_MODULE=$(TCTI_DEFAULT_MODULE) \
-                            -DESYS_TCTI_DEFAULT_CONFIG=$(TCTI_DEFAULT_CONFIG)
-esapi_libesapi_la_LIBADD  = $(libsapi) $(libmarshal) $(libtcti_device) $(libtcti_socket) $(libutil)
-esapi_libesapi_la_LDFLAGS = $(AM_LDFLAGS)  -lgcrypt
-esapi_libesapi_la_SOURCES = $(ESAPI_SRC)
-endif
+### TCG TSS TCTI spec libraries ###
+# tcti headers
+libtctidir      = $(includedir)/tcti
+libtcti_HEADERS = $(srcdir)/include/tcti/*.h
 
-tcti_libtcti_device_la_CFLAGS   = $(AM_CFLAGS)
+# tcti device library
+lib_LTLIBRARIES += libtcti-device.la
+
+libtcti_device_la_CFLAGS   = $(AM_CFLAGS)
 if HAVE_LD_VERSION_SCRIPT
-tcti_libtcti_device_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/tcti/tcti_device.map
+libtcti_device_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/tcti/tcti_device.map
 endif # HAVE_LD_VERSION_SCRIPT
-tcti_libtcti_device_la_LIBADD   = $(libmarshal) $(libutil)
-tcti_libtcti_device_la_SOURCES  = tcti/tcti_device.c
+libtcti_device_la_LIBADD   = $(libmarshal) $(libutil)
+libtcti_device_la_SOURCES  = tcti/tcti_device.c
```

Comment:
Changing this from `tcti_libtcti_device_la` to `libtcti_device_la` moves the location of the `libtcti-device.la` file and where the `libtcti-device.so` will end up. Was this intentional? If so some justification would be helpful.

Reply:
Sorry, that was testing stuff... Reverted to original...